src/libcollections/vec.rs

```diff
@@ -1505,6 +1505,22 @@ impl<'a> From<&'a str> for Vec<u8> {
 // Clone-on-write
 ////////////////////////////////////////////////////////////////////////////////
 
+#[stable(feature = "rust1", since = "1.7.0")]
+impl<'a, T: Clone> From<&'a [T]> for Cow<'a, [T]> {
+    #[inline]
```

Comment:
Those actually need the `#[inline]` because they won't be monomorphized otherwise (e.g. they're concrete and have no type parameters). The purpose there is just to make sure that cross-crate inlining works, but that works already here due to the generics.